### ISSUE TITLE
Document parser breaks quietly when encountering Ellipsis (...)

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1547]
The autodoc parser breaks when it encounters `...` in a file. This breaks adding annotations to class attributes (see attachments).

It gives no error unless `-vv` is enabled, when it produces:

```
[autodoc] => <class 'bad.Bar'>
[autodoc] module analyzer failed: parsing failed (exception was: ParseError("bad input: type=23, value='.', context=('', (2, 7))",))
```

Attached 'bad.py' shows a broken example, attached 'good.py' shows that it works if `...` is changed to `Ellipsis`. Minimal 'conf.py' and 'index.rst' also included.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1547
- Originally reported by: Anonymous
- Originally created at: 2014-08-27T17:25:54.716

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-27 16:14:28+00:00_

I can't reproduce the reported behavior. `Oh no, the docstring is gone!` is displayed as the attached screenshot. Please update your status to reproduce this issue. Thanks!

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-27 16:15:49+00:00_

I attached a console log on my environment (ubuntu-12.04, sphinx-1.2.2, python-2.7)

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-27 16:22:24+00:00_

I got it. It's reproduced with python-3.3.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-28 13:39:10+00:00_

We need update `sphinx.pycode.pgen2.tokenizer` module to fix this because the module doesn't recognize `...` literal (Ellipsis for py3).

I'll fix it for Sphinx-1.3.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-28 13:43:36+00:00_

Fix: pgen2 tokenizer doesn't recognize `...` literal (Ellipsis for py3). Closes #1547

I think pgen2 code derived from lib2to3 package. Basically, the package only support python2 code then it doesn't recognize `...` literal.

→ <<cset f190de75cdcee9a3b88673a615b56bb72527b0c7>>
